src/test/java/examples/websocket/TradeStreamWithAllCallbacks.java

```diff
@@ -0,0 +1,44 @@
+package examples.websocket;
+
+import com.binance.connector.client.impl.WebsocketClientImpl;
+import com.binance.connector.client.utils.WebSocketCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TradeStreamWithAllCallbacks {
+    private static final Logger logger = LoggerFactory.getLogger(TradeStreamWithAllCallbacks.class);
+    private static volatile boolean tradeStreamIsUp = false;
+    private static WebSocketCallback onOpenCallback;
+    private static WebSocketCallback onMessageCallback;
+    private static WebSocketCallback onClosingCallback;
+    private static WebSocketCallback onFailureCallback;
+
+    public static void main(String[] args) {
+        WebsocketClientImpl client = new WebsocketClientImpl();
+        onOpenCallback = openEvent -> {
+            tradeStreamIsUp = true;
+        };
+        onMessageCallback = (messageEvent) -> {
+            System.out.println(messageEvent);
+            client.closeAllConnections();
+        };
+        onClosingCallback = closingEvent -> {
+            tradeStreamIsUp = false;
+            connectToTradeStream(client, onOpenCallback, onMessageCallback, onClosingCallback, onClosingCallback);
```

Comment:
Doesn't this form an infinite loop in the program? You tried to closeAllConnections in the messageCallback but then connectToTradeStream once again.

Reply:
You're right! I've copy pasted the original body of the onMessage.

But it should definitely be removed.

I'll keep only the sysyout msg in onMessage.

Is it ok for you?

Reply:
Yes, that is fine.